### DISCUSSION TITLE
Do I actually need a Service and a Deployment?

### DISCUSSION THREAD

**Discussion by mikebell90** [istio/istio#49649] [Q&A] [answered]
Ok, that sounds ludicrous. Let me explain

For Teamcity agents, they normally launch as a bare pod. Yes we can change the podspec and hook up a serviceacount. So Authz is set. Normally these would only call outgoing, no one would call incoming.

Basic tests seem to work, but are we mucking up anything by NOT creating the Service and Deployment wrappers in this case?

**Comment by howardjohn** [comment, marked as answer]
Telemetry can be a bit wonky for the `workload` labels I think (that is, it will be the pod name which maybe is constantly changing), but otherwise should be good. I would make sure you are on 1.20, there were fixes recently around things that don't have a Service

**Reply by mikebell90** [reply]
Interesting. Is there an annotation or label to disable the prometheus telemetry? Generally speaking I can only see it causing cardinality issues

**Reply by howardjohn** [reply]
Scraping is configured by prometheus itself. Often with `prometheus.io/scrape: false` but its up to your cprometheus config generally
